draft-ietf-mls-protocol.md

```diff
@@ -1000,6 +1000,70 @@ struct {
 } UserInitKey;
 ~~~~~
 
+# Message Framing
+
+[[ TODO prose ]]
+
+~~~~~
+enum {
+    invalid(0),
+    handshake(1),
+    application(2),
+    (255)
+} ContentType;
+
+struct {
+    opaque content[MLSPlaintext.length];
+    uint8 signature[MLSInnerPlaintext.sig_len];
+    uint16 sig_len;
+    ContentType type;
+    uint8 zero\_padding[length\_of\_padding];
+} MLSInnerPlaintext;
+
+struct {
+    opaque group_id<0..255>;
+    uint32 epoch;
+    uint32 sender;
+    ContentType type;
+
+    select (MLSPlaintext.type) {
+        case handshake:   Handshake;
+        case application: Application;
+    }
+
+    opaque signature<1..2^16-1>;
```

Comment:
I am confused now, why is there a second signature here ?
In both cases there is a signature that covers the content of `group_id`, `epoch`, `sender` and `content type` no ?

Reply:
I think this is the only signature we need. Then we can leave out the ContentType and the signature in the inner plaintext.

Reply:
On the contrary, the inner one is the one we use, you don't want to sign a large padding.
What is important is that the inner one must cover those fields as done the message protection section.

Reply:
Ok, then we're assuming that `group_id`, `sender` and `epoch` are still part of the `Handshake` struct? I was wondering if we could somehow avoid redundancy here.